metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_schema.py

```diff
@@ -358,3 +438,97 @@ def get_fk_constraints_for_schema(
 
         return constraints
+
+    def get_tags_for_database_without_propagation(
+        self,
+        db_name: str,
+    ) -> _SnowflakeTagCache:
+        cur = self.query(
+            SnowflakeQuery.get_all_tags_in_database_without_propagation(db_name)
+        )
+
+        tags = _SnowflakeTagCache()
+
+        for tag in cur:
+            snowflake_tag = SnowflakeTag(
+                database=tag["TAG_DATABASE"],
+                schema=tag["TAG_SCHEMA"],
+                name=tag["TAG_NAME"],
+                value=tag["TAG_VALUE"],
+            )
+
+            # This is the name of the object, unless the object is a column, in which
+            # case the name is in the `COLUMN_NAME` field.
+            object_name = tag["OBJECT_NAME"]
+            # This will be null if the object is a database or schema
+            object_schema = tag["OBJECT_SCHEMA"]
+            # This will be null if the object is a database
+            object_database = tag["OBJECT_DATABASE"]
+
+            domain = tag["DOMAIN"]
+            if domain == "DATABASE":
+                tags.add_database_tag(object_name, snowflake_tag)
+            elif domain == "SCHEMA":
+                tags.add_schema_tag(object_name, object_database, snowflake_tag)
+            elif domain == "TABLE":  # including views
+                tags.add_table_tag(
+                    object_name, object_schema, object_database, snowflake_tag
+                )
+            elif domain == "COLUMN":
+                column_name = tag["COLUMN_NAME"]
+                tags.add_column_tag(
+                    column_name,
+                    object_name,
+                    object_schema,
+                    object_database,
+                    snowflake_tag,
+                )
+            else:
+                # This should never happen.
+                self.logger.error(f"Encountered an unexpected domain: {domain}")
+                continue
+
+        return tags
+
+    def get_tags_for_object(
+        self,
+        domain: str,
+        quoted_identifier: str,
+        db_name: str,
+    ) -> List[SnowflakeTag]:
+        tags: List[SnowflakeTag] = []
+
+        cur = self.query(
+            SnowflakeQuery.get_all_tags_on_object(db_name, quoted_identifier, domain),
+        )
+
+        for tag in cur:
+            tags.append(
+                SnowflakeTag(
+                    database=tag["TAG_DATABASE"],
+                    schema=tag["TAG_SCHEMA"],
+                    name=tag["TAG_NAME"],
+                    value=tag["TAG_VALUE"],
+                )
+            )
+        return tags
+
+    def get_tags_on_columns_for_table(
+        self, quoted_table_name: str, db_name: str
+    ) -> Dict[str, List[SnowflakeTag]]:
+        tags: Dict[str, List[SnowflakeTag]] = defaultdict(list)
+        cur = self.query(
+            SnowflakeQuery.get_tags_on_column(db_name, quoted_table_name),
```

Comment:
does this get tags for all columns in the table? if so, can we rename to plural e.g. `get_tags_on_columns`